sympy/physics/vector/vector.py

```diff
@@ -453,6 +453,29 @@ def _det(mat):
     __rand__ = __and__
     __rmul__ = __mul__
 
+    def components(self):
```

Comment:
Instead of adding a separate private function, can't we add this as an attribute to vector class? @moorepants

Reply:
My suggestion is that this components thing is not something the user needs, so we should hide it from the user as an internal private method.

Reply:
This function isn't doing anything unique, just using vector args. So in my opinion we should just simply add an attribute to vector class which returns the components as a dictionary.

```
>>> v = v = 3*R.x + 4*R.y + 5*R.z
>>> v.comp
{R.x: 3, R.y: 4, R.z: 5}

```

Reply:
Sure, but private `_comp`.

Reply:
Yeah I have made it private now .In the vector module `components()` is a public function ( which was my motivation to make it similar ) . I think it is fine as not accessible to the user .But if it should be added as an attribute to vector class, I'll make changes in upcoming commits .Thanks !